src/compiler/checker.ts

```diff
@@ -19387,13 +19400,7 @@ namespace ts {
                     else {
                         // conditionals aren't related to one another via distributive constraint as it is much too inaccurate and allows way
                         // more assignments than are desirable (since it maps the source check type to its constraint, it loses information)
-                        const distributiveConstraint = hasNonCircularBaseConstraint(source) ? getConstraintOfDistributiveConditionalType(source as ConditionalType) : undefined;
-                        if (distributiveConstraint) {
-                            if (result = isRelatedTo(distributiveConstraint, target, RecursionFlags.Source, reportErrors)) {
-                                resetErrorInfo(saveErrorInfo);
-                                return result;
-                            }
-                        }
+                        doDistributiveCheck = true;

```

Comment:
This just move checking of the distributive constraint to _after_ checking the normal constraint, which is needed to remove an instantiation depth error from the example repo in